src/Network/WebSockets/Hybi13/Mask.hs

```diff
@@ -24,11 +25,17 @@ type Mask = Maybe B.ByteString
 --------------------------------------------------------------------------------
 -- | Apply mask
 maskPayload :: Mask -> BL.ByteString -> BL.ByteString
-maskPayload Nothing     = id
-maskPayload (Just mask) = snd . BL.mapAccumL f (cycle $ B.unpack mask)
+maskPayload Nothing = id
+maskPayload (Just "\x00\x00\x00\x00") = id
+maskPayload (Just mask) =
+  BL.fromChunks . reverse . fst . BL.foldlChunks f ([], cycle (B.unpack mask))
```

Comment:
I don't think this is sufficiently lazy.  I would like this function to work on lazy ByteStrings that do not necessarily fit in memory.